lib/components/Attachments/AttachmentCard.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from "react";
+
+import saveAs from "file-saver";
+import { MenuVertical, File } from "neetoicons";
+import { Dropdown, Input, Toastr, Typography, Tooltip } from "neetoui";
+import { assoc } from "ramda";
+
+import directUploadsApi from "apis/direct_uploads";
+
+import { ATTACHMENT_OPTIONS } from "./constants";
+import { validFileName } from "./utils";
+
+const { Menu, MenuItem } = Dropdown;
+
+const AttachmentCard = ({ attachment, endpoint, onChange, attachments }) => {
+  const [isRenaming, setIsRenaming] = useState(false);
+  const [newFilename, setNewFilename] = useState("");
+  const [error, setError] = useState("");
+
+  const handleDownload = () => {
+    saveAs(attachment.url, attachment.fileName);
+  };
+
+  const handleRename = async () => {
+    try {
+      const { signedId } = attachment;
+      const payload = {
+        blob: { filename: newFilename },
+      };
+
+      await directUploadsApi.update(endpoint, signedId, payload);
+      setIsRenaming(false);
+      setNewFilename("");
```

Comment:
Move this to `finally` block instead of repeating it in `try` and `catch` blocks.